experimental/packages/opentelemetry-sdk-metrics-base/src/export/MetricReader.ts

```diff
@@ -14,52 +14,123 @@
+   */
   getPreferredAggregationTemporality(): AggregationTemporality {
-    return this._exporter.getPreferredAggregationTemporality();
+    return this._preferredAggregationTemporality;
   }
 
-  async collect(): Promise<void> {
+  /**
+   * Handle once the SDK has initialized this {@link MetricReader}
+   * Overriding this method is optional.
+   */
+  protected onInitialized(): void {
+    // Default implementation is empty.
+  }
+
+  /**
+   * Handle a shutdown signal by the SDK.
+   *
+   * <p> For push exporters, this should shut down any intervals and close any open connections.
+   * @protected
+   */
+  protected abstract onShutdown(): Promise<void>;
+
+  /**
+   * Handle a force flush signal by the SDK.
+   *
+   * <p> In all scenarios metrics should be collected via {@link collect()}.
+   * <p> For push exporters, this should collect and report metrics.
+   * @protected
+   */
+  protected abstract onForceFlush(): Promise<void>;
+
+  /**
+   * Collect all metrics from the associated {@link MetricProducer}
+   */
+  async collect(options: ReaderCollectionOptions): Promise<MetricData[]> {
     if (this._metricProducer === undefined) {
-      throw new Error('MetricReader is not bound to a MeterProvider');
+      throw new Error('MetricReader is not bound to a MetricProducer');
     }
-    const metrics = await this._metricProducer.collect();
 
-    // errors thrown to caller
-    await this._exporter.export(metrics);
+    // Subsequent invocations to collect are not allowed. SDKs SHOULD return some failure for these calls.
+    if (this._shutdown) {
+      throw new Error('Collection is not allowed after shutdown');
+    }
+
+    return await callWithTimeout(this._metricProducer.collect(), options.timeoutMillis ?? 10000);
   }
 
-  async shutdown(): Promise<void> {
+  /**
+   * Shuts down the metric reader, the promise will reject after the specified timeout or resolve after completion.
+   *
+   * <p> NOTE: this operation will continue even after the promise rejects due to a timeout.
+   * @param options options with timeout (default: 10000ms).
+   */
+  async shutdown(options: ReaderForceFlushOptions): Promise<void> {
+    // Do not call shutdown again if it has already been called.
     if (this._shutdown) {
+      api.diag.error('Cannot call shutdown twice.');
       return;
     }
 
+    await callWithTimeout(this.onShutdown(), options.timeoutMillis ?? 10000);
     this._shutdown = true;
-    // errors thrown to caller
-    await this._exporter.shutdown();
   }
 
-  async forceFlush(): Promise<void> {
+  /**
+   * Flushes metrics read by this reader, the promise will reject after the specified timeout or resolve after completion.
+   *
+   * <p> NOTE: this operation will continue even after the promise rejects due to a timeout.
+   * @param options options with timeout (default: 10000ms).
+   */
+  async forceFlush(options: ReaderShutdownOptions): Promise<void> {
```

Comment:
```suggestion
  async forceFlush(options: ReaderForceFlushOptions): Promise<void> {
```

Reply:
Thanks for spotting this. Fixed in 08501aa